### PR TITLE
Add Basic Functionality Test

### PR DESCRIPTION
**Pull request by shanemacbride**
Implemented `/adop test basic`:
  - runs `/adop compose down --volumes`
  - runs `/adop compose init --without-load`
  - triggers `Load_Platform` job to run
  - checks the states of containers that are not running
  - outputs the results (Load_Platform build result, container names and statuses that are not running)

Example Output from `/adop test basic`:
```
#######################
#  Testing Results:   #
#######################


Load_Platform Build Result: SUCCESS

Docker Containers with Non-running Statuses:
registry: Restarting (1) About a minute ago

```

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Some additional items to implement added to JIRA.

**Comment by shanemacbride** [comment]
#### Requested Changes Implemented

Example test output:
```
Removing volume ldopdockercompose_test_sensu_rabbitmq_data
Removing volume ldopdockercompose_test_sonar_data
Removing volume ldopdockercompose_test_ldap_static
Removing volume ldopdockercompose_test_sensu_client_conf
Removing volume ldopdockercompose_test_ldap_db
Removing volume ldopdockercompose_test_nginx_config
Removing volume ldopdockercompose_test_sonar_mysql_data
Removing volume ldopdockercompose_test_sonar_extensions
Removing volume ldopdockercompose_test_jenkins_home
Removing volume ldopdockercompose_test_elasticsearch_data
Removing volume ldopdockercompose_test_registry_data
Removing volume ldopdockercompose_test_sonar_logs
Removing volume ldopdockercompose_test_sensu_redis_data
Removing volume ldopdockercompose_test_jenkins_slave_home
Removing volume ldopdockercompose_test_sensu_api_conf
Removing volume ldopdockercompose_test_nexus_sonatype_work
Removing volume ldopdockercompose_test_gerrit_review_site
Removing volume ldopdockercompose_test_git_repos
Removing volume ldopdockercompose_test_nginx_releasenote
Removing volume ldopdockercompose_test_gerrit_mysql_data
Removing volume ldopdockercompose_test_registry_certs
Removing volume ldopdockercompose_test_sensu_server_conf

#######################
#  Testing Results:   #
#######################


Load_Platform Build Result: SUCCESS

Docker Containers with Non-running Statuses:

registry: Restarting (1) 46 seconds ago

...basic test complete.
```

Command usage:
```
usage: adop test basic [<options>]

         Options:
          -w : Execute init without the --without-pull flag.
          -h : Prints this help information.
```

Functionality:

- Using temporary volumes, compose init --without-load --without-pull (unless -w flag)
- Trigger the Load_Platform job, wait for it to build, store the result
- Examine running containers, save name of any that are not Running/Up
- Run compose down and remove the temporary volumes
- Output the results of the test, including the build result of Load_Platform and any container with non-running status prior to compose down

File: cmd/test [hunk 1]
```diff
@@ -20,19 +20,82 @@ help() {
...
+      *)
+        echo "Invalid parameter(s) or option(s)."
+        basic_help
+        exit 1
+        ;;
+    esac
+  done
+
+  echo "...checking if LDOP is currently running..."
+  PS=$(${CONF_DIR}/adop compose ps -q | wc -l)
```
Comment (ghost) [inline cmd/test]: Consider the case where something changes in the compose script as to how it performs standard output which then alters what the correct line count is. Is there a way we can instead parse the State field of the compose command and check for running (or other states where we want to prompt)?

**Review by ghost — CHANGES_REQUESTED** [review]

File: cmd/test (hunk 1 above)
Comment (shanemacbride) [reply cmd/test]: replied in person

File: cmd/test [hunk 2]
```diff
@@ -57,11 +57,15 @@ basic() {
...
+  PS=$(${CONF_DIR}/adop compose ps 2>/dev/null)
+
+  # Get all the output after the last occurance of *-\s
+  AFTER_HEADER=${PS##*----}
 
   if [ $F -eq 0 ]
   then
-    if [ $PS -gt 4 ]
+    # If there are any items in /adop compose ps...
+    if [ "${#AFTER_HEADER}" -gt "0" ]
```
Comment (ghost) [inline cmd/test]: This doesn't handle the case where the platform is paused, as stopping is not a permitted operation on a paused container.

Sample error output: ERROR: for elasticsearch  Cannot stop container 6c6cf14118ae2702ddf066f66805e505e79940a785a7df7ee2cc150549c8cd6a: Container 6c6cf14118ae2702ddf066f66805e505e79940a785a7df7ee2cc150549c8cd6a is paused. Unpause the container before stopping

**Review by ghost — CHANGES_REQUESTED** [review]

File: cmd/test [hunk 3]
```diff
@@ -67,15 +67,15 @@ basic() {
...
+      echo "This test will unpause any paused LDOP containers and shut them down."
+      read -p "Proceed to shut down running and paused containers? (Y/N)   " -n 1 -r
       echo
       if [[ $REPLY =~ ^[Yy]$ ]]
       then
         echo "...proceeding..."
         echo "...stopping LDOP..."
-        echo "/adop compose down"
-        ${CONF_DIR}/adop compose down
+        ${CONF_DIR}/adop compose unpause 2>/dev/null
```
Comment (ghost) [inline cmd/test]: Would like to test for a particular condition to determine whether running a command is necessary as opposed to just forcing it and hiding output. 

For example, parse output to determine whether there are any paused containers, and if so, unpause them. If there are not any paused containers, then there's no need to unpause. Should also allow us to surface output from unpause/down commands which could be useful to either users of this command or logs in a CI/CD system should some unexpected scenario arise.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**